CHANGELOG/v1.8.0.md

```diff
@@ -0,0 +1,519 @@
+  - Machine drain ignores terminating Pods earlier for unreachable Nodes (#10706)
+  - MachineSet: Fix deletion priority to avoid deleting too many machines (#10087)
+  - MachineSet: Deprioritize unknown NodeHealthy conditions for deletion (#10763)
+  - MachineSet: Ensure unhealthy machines get deletion priority (#10755)
+- A usual, significantly improved e2e test framework & coverage (#10083, #10216, #10135, #10498, #10639, #10530)
+- New providers in clusterctl: IONOS Cloud (#10767), Linode (#10471), Tinkerbell (#10487)
+
+## Deprecation Warning
+
+- ClusterClass: Deprecate definitionFrom field (#10841)
+- ClusterClass: Deprecate IP Family builtin variable (#10554)
+- MachineDeployment: Deprecate revision management (#10855)
+- MachineHealthCheck: Deprecate MaxUnhealthy & UnhealthyRange fields (#10853)
+
+Umbrella issue: API changes: #10852
+
+## Changes since v1.7.0
+## :chart_with_upwards_trend: Overview
+- 320 new commits merged
+- 1 breaking change :warning:
+- 23 feature additions ✨
+- 49 bugs fixed 🐛
+
+## :warning: Breaking Changes
+- ClusterClass: Adjust CC & Cluster controller to block on variable conflicts, deprecate definitionFrom (#10841)
+
+## :sparkles: New Features
+- API: Add support negative polarity conditions (#10550)
+- API: Optimize rbac across controllers (#10552)
+- Bootstrap: Allow CAPBK to generate JoinConfiguration discovery kubeconfig (#10799)
+- CABPK: Add ExtraEnvs and ImagePullSerial to KubeadmConfig (#10846)
+- CABPK: Add kubeadm v1beta4 types (#10709)
+- ClusterCacheTracker: Add QPS & burst options & flags for ClusterCacheTracker (#10880)
+- ClusterClass: Add map key for MD/MP class & topology in ClusterClass & Cluster.spec.topology (#10808)
+- ClusterClass: Add metadata for top-level and nested variables & deprecate previous metadata field (#10778)
+- ClusterClass: Add variable validation to ClusterClass controller, block Cluster reconcile if variables not reconciled (#10812)
+- ClusterClass: Extend ControlPlane/ MD / MP builtin to include metadata (#10837)
+- ClusterClass: Introduce CEL for ClusterClass Variables (#9239)
+- ClusterClass: Support ControlPlane variable overrides (#10682)
+- e2e: Bump Kubernetes to v1.31.0-rc.1 (#11022)
+- e2e: Bump Kubernetes version used for testing to v1.30.0-rc.1 (#10384)
+- e2e: Bump Kubernetes version used for testing to v1.31.0-beta.0 (#10922)
+- e2e: Bump Kubernetes version used for testing to v1.31.0-rc.0 (#10968)
+- KCP: Default ControlPlaneKubeletLocalMode feature gate to true for >= 1.31.0 (#10947)
+- MachinePool: MachinePools: Use NodeDeletionTimeout and default it to 10s (#10553)
+- MachinePool: Set Kubernetes version in machinepool machine Status.Version (#10849)
+- MachineSet: Add remediation strategy support in MachineDeployment (#10712)
+- Runtime SDK/ClusterClass: Extend cluster builtin to include UID (#10822)
+- Runtime SDK/ClusterClass: Fix GetObjectVariableInto util func (#10702)
+- Testing: Support e2e for WSL (#10402)
+
+## :bug: Bug Fixes
```

Comment:
Please add https://github.com/kubernetes-sigs/cluster-api/pull/11042

(it will merge before the release)